File: Pregunta1/Haskell/fiboR.hs
fibo :: Int -> Int
fibo 1 = 0
fibo 2 = 1
fibo 3 = 1
fibo n | n > 3 = fibo(n-1) + fibo(n-2) + fibo(n-3)

fibonacci ::Int -> [Int]
fibonacci n = [fibo(x) | x <- [1..n], x > 0]

main = do
 putStrLn "\nFibonacci Recursivo"
 putStrLn "Introduzca un N : "
 n<-getLine
 putStrLn "Los N primeros fibonacci "
 print(fibonacci(read n :: Int))

File: Pregunta1/Scala/fiboR.scala
object fiboE{
	def fibo (n : Int) : Int ={
		if (n == 1) return 0
		else if (n == 2) return 1
		else if (n == 3) return 1
		else return fibo(n-1) + fibo(n-2) + fibo(n-3) 
	} 

	def main(args:Array[String]):Unit={
		println("Fibonacci Recursivo")
		print("Introduzca un N : ")
		val n = scala.io.StdIn.readInt()
		var i = 1
		while(i <= n){
			print(fibo(i))
			print(", ")
			i = i + 1
		}
	}
}